src/eterna/constraints/ConstraintBar.ts

```diff
@@ -40,59 +73,181 @@ export default class ConstraintBar extends ContainerObject {
     }
 
     protected added() {
-        for (let constraint of this._constraints) {
-            this.addObject(constraint.constraintBox, this.container);
-            constraint.constraintBox.pointerDown.connect(() => {
-                this.onConstraintBoxClicked(constraint);
-            });
+        const {config} = ConstraintBar;
+
+        this._constraintsLayerRoot = new Container();
+
+        const drawerEnabled = this._constraints.length > 1;
+        if (drawerEnabled) {
+            // Background
+            this._background = (() => {
+                const bg = new Graphics();
+                this.container.addChild(bg);
+                bg.interactive = true;
+                bg.on('pointerdown', (e) => {
```

Comment:
It looks like a better way, but it's difficult to implement that functionality in a subclass, there are state variables like `_backgroundDrag` needed to allow scrolling the constraints when sliding either the constraints or the background. Ideally, the constraints bar need to handle input in one centralized location (a foreground layer) and from there, knowing the constraints hitboxes, determine whether to trigger a scroll, or a constraint click, or a constraint hover. Currently, the constraints handle their own input which makes it harder to handle global actions like "scroll" alongside individual ones like "click" and "hover". Let me know if you have a better solution for this!

Reply:
Ah, I see! Didn't realize that it was modifying class variables. You could create signals for all of these things, but I don't think you really gain anything one way or the other.

Hm, well... If you did it that way, you could handle it like:
```ts
bg.pointerDown.connect(() => {
  // ...
});
bg.pointerMove.connect(...)
// ...
```

Thoughts about handling it that way? I would even go so far as adding a GraphicsObject class to flashbang, so that you can just:
`let bg = new GraphicsObject();`

Reply:
Ah I see, yes that is better indeed. Keep the signals implementation inline, but create a GraphicsObject instance. Will do!